src/metabase/lib/js.cljs

```diff
@@ -872,26 +872,41 @@
 (def ^:private row-cell       (js-cells-by #(.-col ^js %)))
 (def ^:private dimension-cell (js-cells-by #(.-column ^js %)))
 
+(defn- drill-thru-context [a-query stage-number column value row dimensions]
+  (lib.convert/with-aggregation-list (lib.core/aggregations a-query stage-number)
+    (let [column-ref (when-let [a-ref (.-field_ref ^js column)]
```

Comment:
Maybe out of scope: Maybe I misunderstand where these structures are coming from but if the FE is storing these in `data-attributes` or something, it seems like too much knowledge to know that `column` has a `field-ref` key, same for `row-cell .-col`  and `dimension-cell .-column`. Should the caller not pass in the data we're interested in rather than encoding what seems to be a rather arbitrary js datastructure. If it's round-tripped from BE, can they use an opaque object?

Regardless, would appreciate a comment where these objects are coming from and expected shapes.

Reply:
This is definitely out of scope; this code is not new and has been part of `available-drill-thrus` since its creation.

On the one hand, I agree this kinda sucks - coupling these calls to how the FE stores data about series, dimensions, rows of data, etc. Some of this data is coming directly from the BE, and some is derived in the FE. I suspect someday we'll see it refactored end to end, and that's a good thing. But this is not that day. This file is here to bridge between the FE and `metabase.lib.*`, so there's nowhere better for it to live.[1]

[1] It could also live in the TS wrapper, all else being equal. But getting passed the original FE objects is handy, because they're now being used as cache keys, effectively by `===` identity, so they need to be identical objects across calls for the caching to help.